agimus_demo_04_visual_servoing/agimus_demo_04_visual_servoing/reference_publisher.py

```diff
@@ -45,43 +44,67 @@ def __init__(self):
         self.tf_buffer = Buffer()
         self.tf_listener = TransformListener(self.tf_buffer, self)
         self.tf_static_broadcaster = StaticTransformBroadcaster(self)
+        self.create_subscription(
+            PoseStamped, "/object/detections", self.apriltag_callback, 5
+        )
 
         # This frame should be the same as the one used in agimus_controller
         # visual servoing residual
-        self.camera_frame = "camera_color_optical_frame"
+        self.world_frame = "fer_link0"
+        self.end_effector_frame = "fer_hand_tcp"
         self.object_frame = "current_object"
 
+    def apriltag_callback(self, pose_msg):
```

Comment:
```suggestion
    def vision_callback(self, pose_msg):
```

It took me some time to understand that this is not directly the result of apriltag because apriltag returns the transform in the camera frame.